src/backend/marsha/core/lti/utils.py

```diff
@@ -0,0 +1,91 @@
+"""Helpers to create a dedicated resources."""
+from django.db.models import Q
+
+from pylti.common import LTIException
+
+from ..defaults import PENDING, READY
+from ..models import Playlist
+
+
+def get_or_create_resource(model, lti):
+    """Get or Create a resource targeted by a LTI request.
+
+    This function is generic and will use the `model` argument to create
+    the wanted resource.
+
+    Create the playlist if it does not pre-exist (it can only happen with consumer site scope
+    passports).
+
+    Parameters
+    ----------
+    lti : Type[LTI]
+
+    model:
+        The model we want to get or create.
+
+    Raises
+    ------
+    LTIException
+        Exception raised if the request does not contain a context id (playlist).
+
+    Returns
+    -------
+    An instance of the model targeted by the LTI request or None
+
+    """
+    lti.verify()
+    try:
+        assert lti.context_id
+    except AssertionError:
+        raise LTIException("A context ID is required.")
+
+    # If the video already exists, retrieve it from database
+    filter_kwargs = {"upload_state": READY} if lti.is_student else {}
```

Comment:
I think this is dangerous in case a new role is added tomorrow for example. It is safer to list the roles that are allowed:
```suggestion
    filter_kwargs = {"upload_state": READY} if (lti.is_instructor or lti.is_administrator) else {}
```

Reply:
Yes sorry ^^